doc/manual/src/release-notes/rl-next.md

```diff
@@ -52,3 +52,12 @@
 
```

Comment:
Another problem I just discovered: `baseNameOf` return the virtual path string for lazy path roots:

```
nix-repl> baseNameOf lazyPath
"virtual0000000000000000000000004-source"
```

It should really throw an error, it doesn't make sense to get the base name of a lazy path.

Reply:
> It should really throw an error, it doesn't make sense to get the base name of a lazy path.

This is again similar to the basename of the filesystem root.

```
nix-repl> baseNameOf /. 
""

```

It would be more consistent for `baseNameOf lazy` to return `""`, but I'm not sure which is better.

Reply:
I just found [a piece of code](https://github.com/NixOS/nixpkgs/pull/188301#discussion_r969452035) that would use `baseNameOf` on flake paths and whose behavior would change silently with this PR. While there's problems with that piece of code (inspecting store paths isn't great), and this was [pointed out](https://github.com/NixOS/nixpkgs/pull/188301#discussion_r1073997929), it still shows how easy it is for this PR to break code in unintentional ways. I still think `baseNameOf` should give an error or as suggested by @roberth at least be consistent with `baseNameOf /. == ""`. This also relates to my thinking that lazy paths shouldn't be `toString`'able at all: https://github.com/NixOS/nix/pull/6530#discussion_r1029941664